ledger-api/grpc-definitions/com/daml/ledger/api/v1/active_contracts_service.proto

```diff
@@ -56,9 +58,59 @@ message GetActiveContractsResponse {
 
   // The workflow that created the contracts.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional, In the Ledger API 3.0 this field will be removed.
   string workflow_id = 2;
 
   // The list of contracts that were introduced by the workflow with ``workflow_id`` at the offset.
   // Must be a valid LedgerString (as described in ``value.proto``).
+  // In the Ledger API 3.0 this field will be removed.
+  // This is empty for multi-domain API.
   repeated CreatedEvent active_contracts = 3;
+
+  // Non-empty for multi-domain API.
+  repeated ContractState contract_states = 4;
+}
+
+// In the entire stream of ContractState-s there can be multiple ContractState messages belong to the same contract.
+message ContractState {
```

Comment:
somehow ContractState sounds even more confusing now that it's just one part of the state. Don't really have any great alternatives unfortunately.

Reply:
I tried "ContractStateComponent"...but still not happy. Open for suggestions